### PR TITLE
Merge StatusChange with Update

### PR DESCRIPTION
**Pull request by PoignardAzur**

### CODE REVIEW

**Review by jaredoconnell — APPROVED** [review]
Tested. It works. As long as this doesn't cause any ordering issues with regards to other events, I don't see any disadvantage to this change.

File: masonry/src/widget/button.rs [hunk 1]
```diff
@@ -111,9 +111,19 @@ impl Widget for Button {
...
+        match event {
+            Update::HoveredChanged(_) => {
+                ctx.request_paint_only();
+            }
+            Update::FocusChanged(_) => {
+                ctx.request_paint_only();
+            }
+            Update::DisabledChanged(_) => {
+                ctx.request_paint_only();
+            }
```
Comment (Philipp-M) [inline masonry/src/widget/button.rs:124]:
```suggestion
            Update::HoveredChanged(_) | Update::FocusChanged(_) | Update::DisabledChanged(_) => {
                ctx.request_paint_only();
            }
```

File: masonry/src/widget/checkbox.rs [hunk 2]
```diff
@@ -102,9 +102,19 @@ impl Widget for Checkbox {
...
+        match event {
+            Update::HoveredChanged(_) => {
+                ctx.request_paint_only();
+            }
+            Update::FocusChanged(_) => {
+                ctx.request_paint_only();
+            }
+            Update::DisabledChanged(_) => {
+                ctx.request_paint_only();
+            }
```
Comment (Philipp-M) [inline masonry/src/widget/checkbox.rs:115]:
```suggestion
            Update::HoveredChanged(_) | Update::FocusChanged(_) | Update::DisabledChanged(_) => {
                ctx.request_paint_only();
            }
```

Comment (Philipp-M) [reply masonry/src/widget/checkbox.rs:115]: Or even better (same for the other suggestion):

```rust
        if matches!(
            event,
            Update::HoveredChanged(_) | Update::FocusChanged(_) | Update::DisabledChanged(_)
        ) {
            ctx.request_paint_only();
        }
```

**Pull request merged**